### ISSUE TITLE
Dell Precision 5520: ERROR: Could not execute SMI.

### ISSUE THREAD

**Issue by cmuell89** [dell/libsmbios#18]
Dell Precision 5520 user with Ubuntu 16.04. After the most recent BIOS update (v 1.2.4
), I no longer can use smbios utilities. I receive the following output  for all smbios commands using smbios-thermal-ctl as an example:

```sudo smbios-thermal-ctl -i
Libsmbios version : 2.3.0
smbios-thermal-ctl version : 2.3.0

 Print all the Available Thermal Information of your system: 
-------------------------------------------------------------------
ERROR: Could not execute SMI.

Common problems are:

    -- Insufficient permissions to perform operation.
       Try running as a more privileged account.
          Linux  : run as 'root' user
          Windows: run as 'administrator' user

    -- dell_rbu or dcdbas device drivers not loaded.
       Try loading the dell_rbu driver
          Linux  : modprobe dell_rbu
          Windows: dell_rbu driver not yet available.
```

I have run the command as root to no avail and tried loading the dell_rbu driver to no avail. I have purged and reinstalled libsmbios-dev with no luck.

Also, I am wondering to what extent the thermal mode has on fan usage. I am trying to lessen the amount the fans turn on. They are rarely ever off entirely. After some repeated heavy use, they linger at 3200 rpm for a long time before dropping down to 2500, despite CPU temps below 40C.

Thanks.

**Comment by superm1** [comment]
Unfortunately this is looking like a BIOS regression, not a libsmbios bug.

Two mitigations that I can offer until it's fixed in a new BIOS release:
1. Turn on legacy option ROMs in BIOS setup.  This will require turning off secure boot if you have it enabled.
2. Revert to an older BIOS release.  You can download the executable from support.dell.com.
Flash using the F12 setup menu (directions here: http://www.dell.com/support/article/us/en/19/SLN171755/updating-the-dell-bios-in-linux-and-ubuntu-environments?lang=EN#updatebios2015)

**Comment by cmuell89** [comment]
@superm1 
Thanks for the reply. I'll close this issue. However, a few questions:

Is there a place I might best report this BIOS regression? 

Should the smbios-thermal-ctl have much an effect on fans profiles managed by the BIOS?

I'd prefer to keep the BIOS as the previous BIOS had issues with stressing the fans (loud screech and fans go full blast) during a restart event. 

Cheers

**Comment by superm1** [comment]
@cmuell89  It would be best if you could report it through ProSupport.  The more customers reporting it to support the sooner this could be escalated and fixed.  You can also report it on the Dell forums, but a direct report to ProSupport would be better.

To your question, it should only let you pick the different profiles, but I believe this shouldn't be exposing anything different than you can already get from BIOS setup.  You should post to the Dell forums for questions about the fans being on constantly, but I'll give you some general tips that might help:
* If you're not using the discrete GPU typically, turn it off.
* The Dell factory image should be fairly well tuned for a balance of performance, power and thermal performance.  If you're not using the Dell factory Ubuntu image, make sure you at least install TLP and configure it with some sane defaults.
* Run `powertop `and optimize anything still missing to see if you can get any other improvements.  The more time the CPU is resident in C10, and the PCH in PC10 the better thermal performance you'll be getting as a result.

**Comment by cmuell89** [comment]
@superm1 Thanks for the info. I currently use the factory image but also run powertop and tlp. I actually reported an Ubuntu bug where, at least for the Precision 5520, C-States lower than C3 were not being used. It's been included in the latest kernel :)

I appreciate you replies. I'll close this issue.

**Comment by cmuell89** [comment]
Just FYI. This has been resolved with the Dell BIOS 1.3.3 as of 5/22/17.

**Comment by superm1** [comment]
@cmuell89 Thanks for that confirmation.
